File: online-judge-solutions/Codeforces/1689A.cpp
//https://codeforces.com/problemset/problem/1689/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	100 KB
#include<iostream>
#include<algorithm>
using namespace std;
#define fastInp cin.tie(0); cout.tie(0); ios_base::sync_with_stdio(0);
int main(){
	fastInp;
	int t,n,m,k,ma,mb;
	cin>>t;
	while(t--){
		cin>>n>>m>>k;
		string a,b,c; cin>>a>>b;
		sort(begin(a),end(a),greater<char>());
		sort(begin(b),end(b),greater<char>());
		ma=0;mb=0;
		while(a.size()>0 && b.size()>0){
			bool g=a.back()<b.back();
			if(g && ma==k) g=0;
			if(!g && mb==k) g=1;
			if(g) c+=a.back(),ma++,mb=0,a.pop_back();
			else c+=b.back(),mb++,ma=0,b.pop_back();
		}
		cout<<c<<endl;
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1689A.py
#https://codeforces.com/problemset/problem/1689/A
#Lang		:	Python 3.8
#Time		:	61 ms
#Memory         :	0 KB
for _ in range(int(input())):
    n,m,k=map(int,input().split())
    a,b,c,ma,mb=input(),input(),"",0,0
    a=''.join(sorted(a,reverse=True))
    b=''.join(sorted(b,reverse=True))
    while(len(a) and len(b)):
        g=a[-1]<b[-1]
        if(g and ma==k):g=0
        if(not g and mb==k):g=1
        if(g):
            c+=a[-1]
            ma+=1
            mb,a=0,a[:-1]
        else:
            c+=b[-1]
            mb+=1
            ma,b=0,b[:-1]
    print(c)
